pkg/model/iam.go

```diff
@@ -182,6 +205,64 @@ func (b *IAMModelBuilder) Build(c *fi.ModelBuilderContext) error {
 	return nil
 }
 
+// buildSharedIAMTasks adds a shared IAM tasks to the ModelBuilderContext
+func (b *IAMModelBuilder) buildSharedIAMTasks(id *string, c *fi.ModelBuilderContext) error {
```

Comment:
@chrislovecnm I've updated this to create the additional IAM tasks as well.

Since theres a lot of duplicate logic between this and the regular IAM task creation, I considered making this a function that accepts a `shared bool` and a role name (either `b.IAMName(role)` if managed or `findCustomAuthNameFromArn(roleArn)` if shared). This would create these shared tasks with the same policy documents and such that the managed tasks would have. I'm not sure if thats desirable or if we'd just be ignoring those values in the `Find` function of each shared task.

Thoughts?